metricbeat/module/beats/beats.go

```diff
@@ -0,0 +1,64 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package beats
+
+import (
+	"encoding/json"
+	"net/url"
+
+	"github.com/elastic/beats/metricbeat/helper"
+)
+
+// ModuleName is the name of this module.
+const ModuleName = "beats"
+
+// Info construct contains the data from the Beats / endpoint
+type Info struct {
+	UUID string `json:"uuid"`
+	Beat string `json:"beat"`
+}
+
+// GetInfo returns the data for the Beats / endpoint.
+func GetInfo(m *MetricSet) (*Info, error) {
+	content, err := fetchPath(m.HTTP, "/", "")
+	if err != nil {
+		return nil, err
+	}
+
+	info := &Info{}
+	err = json.Unmarshal(content, &info)
+	if err != nil {
+		return nil, err
+	}
+
+	return info, nil
+}
+
+func fetchPath(http *helper.HTTP, path string, query string) ([]byte, error) {
```

Comment:
I know it's a WIP but I was peeking around... :slightly_smiling_face: Maybe you'll like to call `http` variable something like `httpHelper` or similar so it cannot be confused with `net/http` Go package. At least I was confused at first glance because I couldn't remember any `SetURI` function in the package :smile: